llvm/lib/Target/AMDGPU/AMDGPUInstructionSelector.cpp

```diff
@@ -3662,13 +3666,19 @@ bool AMDGPUInstructionSelector::selectGlobalLoadLds(MachineInstr &MI) const{
   if (isSGPR(Addr))
     MIB.addReg(VOffset);
 
-  MIB.add(MI.getOperand(4))  // offset
-     .add(MI.getOperand(5)); // cpol
+  MIB.add(MI.getOperand(4)); // offset
+
+  bool IsGFX12Plus = AMDGPU::isGFX12Plus(*Subtarget);
```

Comment:
I really hate generation checks. But at least could directly check Subtarget->getGeneration instead of using the weird MC helper function